File: codes/Data_Cleaning.R
# Required packages
install.packages(c("dplyr", "tidyr", "readr", "tm", "Matrix"))
library(dplyr)
library(tidyr)
library(readr)
library(tm)
library(Matrix)

# Load the CSV into a data frame
df <- read_csv('/Users/qbs/downloads/games.csv')

# Initial inspection
print(head(df))

# Define clean_price function
clean_price <- function(price) {
  ifelse(is.na(as.numeric(price)), NA, as.numeric(price))
}

df$Price <- sapply(df$Price, clean_price)

# Create price category
df <- df %>%
  mutate(Price_Category = cut(Price, breaks=c(-1, 0, 10, Inf), labels=c('Free', 'Low-cost', 'Expensive')))

# Only quantitative columns
df_quantitative <- df %>%
  select_if(is.numeric)

# Only qualitative columns
df_qualitative <- df %>%
  select_if(is.character)

print(head(df))

# Text vectorization
df <- df %>%
  filter(!is.na(Name))

corpus <- Corpus(VectorSource(df$Name))
dtm <- DocumentTermMatrix(corpus)
df_vectorized <- as.data.frame(as.matrix(dtm))
print(head(df_vectorized))

# Drop non-useable columns
columns_to_drop <- c('Reviews', 'Support url', 'Support email', 'Metacritic score',
                     'Metacritic url', 'User score', 'Score rank', 'Notes', 'Screenshots', 'Movies')
df <- df %>%
  select(-one_of(columns_to_drop))

# Handle missing values
df <- df %>%
  mutate(`About the game` = ifelse(is.na(`About the game`), 'No Description', `About the game`),
         `Supported languages` = ifelse(is.na(`Supported languages`), 'No Supported Languages', `Supported languages`),
         `Full audio languages` = ifelse(is.na(`Full audio languages`), 'No Audio Languages', `Full audio languages`),
         Website = ifelse(is.na(Website), 'No Websites', Website),
         Developers = ifelse(is.na(Developers), 'No Developers', Developers),
         Publishers = ifelse(is.na(Publishers), 'No Publishers', Publishers),
         Categories = ifelse(is.na(Categories), 'No Categories', Categories),
         Genres = ifelse(is.na(Genres), 'No Genres', Genres),
         Tags = ifelse(is.na(Tags), 'No Tags', Tags)) %>%
  drop_na()

print(head(df))

File: codes/Data_Cleaning.py
import pandas as pd

# Load the CSV into a DataFrame
df = pd.read_csv('/Users/qbs/downloads/games.csv')

# Initial inspection
print(df.head())

def clean_price(price):
    try:
        return float(price)
    except:
        return None 

df['Price'] = df['Price'].apply(clean_price)

bins = [-1, 0, 10, 1000] 
labels = ['Free', 'Low-cost', 'Expensive']
df['Price_Category'] = pd.cut(df['Price'], bins=bins, labels=labels)

# Only quantitative columns
df_quantitative = df.select_dtypes(include=['float64', 'int64'])

# Only qualitative columns
df_qualitative = df.select_dtypes(exclude=['float64', 'int64'])
print(df.head())

from sklearn.feature_extraction.text import CountVectorizer
# Remove rows with NaN 'Name' values
df = df[df['Name'].notna()]

# OR fill NaN with a placeholder
# df['Name'].fillna('unknown', inplace=True)

vectorizer = CountVectorizer()
X = vectorizer.fit_transform(df['Name'])
df_vectorized = pd.DataFrame(X.toarray(), columns=vectorizer.get_feature_names_out())

print(df_vectorized.head())

#Drop non-useable columns
# List of columns to drop
columns_to_drop = ['Reviews', 'Support url', 'Support email', 'Metacritic score', 
                   'Metacritic url', 'User score', 'Score rank', 'Notes', 'Screenshots', 'Movies']

# Drop the columns
df = df.drop(columns=columns_to_drop, errors='ignore')

#Handle missing values
df['About the game'] = df['About the game'].fillna('No Description')
df['Supported languages'] = df['Supported languages'].fillna('No Supported Languages')
df['Full audio languages'] = df['Full audio languages'].fillna('No Audio Languages')
df['Website'] = df['Website'].fillna('No Websites')
df['Developers'] = df['Developers'].fillna('No Developers')
df['Publishers'] = df['Publishers'].fillna('No Publishers')
df['Categories'] = df['Categories'].fillna('No Categories')
df['Genres'] = df['Genres'].fillna('No Genres')
df['Tags'] = df['Tags'].fillna('No Tags')
df = df.dropna()

print(df.head())